openvino/conditional_compilation/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+# ******************************************************************************
+# Copyright 2017-2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ******************************************************************************
+
+set(TARGET_NAME conditional_compilation)
+
+add_library(${TARGET_NAME} INTERFACE)
+
+add_library(openvino::conditional_compilation ALIAS ${TARGET_NAME})
+
+target_link_libraries(${TARGET_NAME} INTERFACE openvino::itt)
+
+target_include_directories(${TARGET_NAME} INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/include)
+
+if(SELECTIVE_BUILD STREQUAL "COLLECT")
+    target_compile_definitions(${TARGET_NAME} INTERFACE -DSELECTIVE_BUILD_ANALYZER)
+elseif(SELECTIVE_BUILD STREQUAL "ON")
+    target_compile_definitions(${TARGET_NAME} INTERFACE -DSELECTIVE_BUILD)
+    ov_force_include(${TARGET_NAME} INTERFACE ${SELECTIVE_BUILD_DEFS})
```

Comment:
where is `SELECTIVE_BUILD_DEFS` defined?

Reply:
It should be provided in CMake arguments. I'm going to hide code generator inside CMake. After that, instead of SELECTIVE_BUILD_DEFS the path to the directory with the collected statistics will be specified.

Reply:
After the code generation for selective build is hidden inside CMake, the SELECTIVE_BUILD_DEFS was replaced by SELECTIVE_BUILD_STAT.